test/unit-tests/services/help-service.ts

```diff
@@ -434,5 +445,72 @@ and another one`
 				assert.isTrue(output.indexOf("index data is read") >= 0);
 			});
 		});
+
+		describe("extensions tests", () => {
+			const assertData = async (expectedEnumerateFilesInDirectorySyncCalledCounter: number, extensionsData?: IExtensionData[]): Promise<void> => {
+				const injector = createTestInjector({ isProjectTypeResult: false, isPlatformResult: true });
+				const $staticConfig = injector.resolve<Config.IStaticConfig>("staticConfig");
+				$staticConfig.MAN_PAGES_DIR = "man_pages_dir";
+				$staticConfig.HTML_PAGES_DIR = "html_pages_dir";
+				$staticConfig.CLIENT_NAME = "client name";
+				const extensionDocsDir = "extensionDocsDir";
+				let enumerateFilesInDirectorySyncCalledCounter = 0;
+				injector.register("fs", {
+					enumerateFilesInDirectorySync: (path: string) => {
+						enumerateFilesInDirectorySyncCalledCounter++;
+						if (path === extensionDocsDir) {
+							return [join(extensionDocsDir, "foo.md")];
+						}
+
+						return [];
+					},
+					readText: (pathToRead: string) => {
+						return pathToRead === join(extensionDocsDir, "foo.md") ? blaEnd : "";
+					},
+					exists: (filePath: string) => true
+				});
+
+				const $extensibilityService = injector.resolve<IExtensibilityService>("extensibilityService");
+				extensionsData = extensionsData || [];
+				extensionsData.push({
+					extensionName: "extension3",
+					docs: extensionDocsDir,
+					version: "1.2.3",
+					pathToExtension: "extension3path"
+				});
+
+				$extensibilityService.getInstalledExtensionsData = (): IExtensionData[] => extensionsData;
+
+				const helpService = injector.resolve<IHelpService>("helpService");
+				await helpService.showCommandLineHelp("foo");
+				const output = injector.resolve("logger").output;
+				assert.isTrue(output.indexOf(blaEnd) >= 0);
+
+				assert.equal(enumerateFilesInDirectorySyncCalledCounter, expectedEnumerateFilesInDirectorySyncCalledCounter,
+					`The enumerateFilesInDirectorySync method must be called exactly ${enumerateFilesInDirectorySyncCalledCounter} times.`);
+			};
+
+			it("help from a single installed extension is successfully loaded", async () => {
+				await assertData(2);
+			});
+
+			it("when multiple extensions are installed, the full ", async () => {
```

Comment:
this test's title has a rather abrupt ending

Reply:
😸